src/symmetric/message_hash/top_level_poseidon.rs

```diff
@@ -210,12 +206,7 @@ impl<
         );
 
         // How many bits can be represented by one field element
-        let bits_per_fe = f64::floor(f64::log2(
-            BigUint::from(FqConfig::MODULUS)
-                .to_string()
-                .parse()
-                .unwrap(),
-        ));
+        let bits_per_fe = f64::floor(f64::log2(F::ORDER_U64 as f64));
 
         // Check that we have enough bits to encode message
         let message_fe_bits = bits_per_fe * f64::from(MSG_LEN_FE as u32);
```

Comment:
I guess we should
* Make 9 (capacity), 15 (rate), and 24 (width) named constants
* add assertion that message+randomness+parameter+epoch+counter do not exceed the width